src/views/docs-view/server.ts

```diff
@@ -139,11 +146,60 @@ export function getStaticGenerationFunctions<
 	// Defining a getter here so that we can pass in remarkPlugins on a per-request basis to collect headings
 	const getLoader = (
 		extraOptions?: Partial<ConstructorParameters<typeof RemoteContentLoader>[0]>
-	) => new RemoteContentLoader({ ...loaderOptions, ...extraOptions })
+	) => {
+		if (isDeployPreview(productSlugForLoader)) {
+			// options needed to read content from the file system when running in preview mode
+			const fsOptions = {
+				localContentDir: `../content/${basePath}`,
+				navDataFile: `../data/${
+					currentRootDocsPath.navDataPrefix ?? currentRootDocsPath.path
+				}-nav-data.json`,
+			}
+
+			/**
+			 * These plugins are run during our content ETL process for remote content, but we need to run
+			 * them when loading content directly from the filesystem.
+			 */
+			const remarkPluginsForFileSystemContent = [
+				[
+					includeMarkdown,
+					{
+						resolveMdx: true,
+						resolveFrom: path.join(process.cwd(), '..', 'content', 'partials'),
+					},
+				],
+				paragraphCustomAlerts,
+			]
+
+			return new FileSystemLoader({
+				...loaderOptions,
+				...fsOptions,
+				...extraOptions,
+				remarkPlugins(params) {
+					const remarkPluginsFromExtraOptions =
+						typeof extraOptions.remarkPlugins === 'function'
+							? extraOptions.remarkPlugins(params)
+							: extraOptions.remarkPlugins
+
+					return [
+						...remarkPluginsFromExtraOptions,
+						...remarkPluginsForFileSystemContent,
+					]
+				},
+			})
```

Comment:
It might be a good idea to abstract out this logic into a separate method, this file is getting pretty long and complex.